Code/Framework/AzToolsFramework/AzToolsFramework/Input/QtEventToAzInputManager.cpp

```diff
@@ -381,25 +398,73 @@ namespace AzToolsFramework
+        if (rect.left() < point.x()) 
+        {
+            point.setX(rect.right() - 1);
+        }
+        else if (rect.right() > point.x()) 
+        {
+            point.setX(rect.left() + 1);
+        }
+    }
+
+    void wrapCursorY(const QRect& rect, QPoint& point) {
+        if (rect.top() < point.y()) 
+        {
+            point.setY(rect.bottom() - 1);
+        }
+        else if (rect.bottom() > point.y()) 
+        {
+            point.setY(rect.top() + 1);
+        }
+    }
+
+
     void QtEventToAzInputMapper::HandleMouseMoveEvent(const QPoint& globalCursorPosition)
     {
         const QPoint cursorDelta = globalCursorPosition - m_previousGlobalCursorPosition;
+        QScreen* screen = m_sourceWidget->screen();
+        const QRect widgetRect(m_sourceWidget->mapToGlobal(QPoint(0,0)), m_sourceWidget->size());
 
         m_mouseDevice->m_cursorPositionData2D->m_normalizedPosition =
             WidgetPositionToNormalizedPosition(m_sourceWidget->mapFromGlobal(globalCursorPosition));
         m_mouseDevice->m_cursorPositionData2D->m_normalizedPositionDelta = WidgetPositionToNormalizedPosition(cursorDelta);
 
-        ProcessPendingMouseEvents(cursorDelta);
-
-        if (m_capturingCursor)
-        {
-            // Reset our cursor position to the previous point
-            AzQtComponents::SetCursorPos(m_previousGlobalCursorPosition);
-        }
-        else
+        switch(m_cursorMode) 
         {
-            m_previousGlobalCursorPosition = globalCursorPosition;
+            case CURSOR_MODE_CAPTURED:
+                AzQtComponents::SetCursorPos(m_previousGlobalCursorPosition);
+                break;
+            case CURSOR_MODE_WRAPPED_X:
+                QPoint screenPos(globalCursorPosition);
+                wrapCursorX(widgetRect, screenPos);
+                QCursor::setPos(screen, screenPos);
+                QPoint screenDelta = globalCursorPosition - screenPos;
+                m_previousGlobalCursorPosition = globalCursorPosition - screenDelta;
+                break;
+            case CURSOR_MODE_WRAPPED_Y:
+                QPoint screenPos(globalCursorPosition);
+                wrapCursorY(widgetRect, screenPos);
+                QCursor::setPos(screen, screenPos);
+                QPoint screenDelta = globalCursorPosition - screenPos;
+                m_previousGlobalCursorPosition = globalCursorPosition - screenDelta;
+                break;
+            case CURSOR_MODE_WRAPPED:
+                QPoint screenPos(globalCursorPosition);
+                wrapCursorX(widgetRect, screenPos);
+                wrapCursorY(widgetRect, screenPos);
+                QCursor::setPos(screen, screenPos);
+                QPoint screenDelta = globalCursorPosition - screenPos;
+                m_previousGlobalCursorPosition = globalCursorPosition - screenDelta;
+                break;
+            default:
+                m_previousGlobalCursorPosition = globalCursorPosition;
```

Comment:
suggestion: We most likely want to Assert here and cover all cases in the enum to be more explicit, we should ideally never get to `default`